addons/product/product.py

```diff
@@ -432,6 +432,16 @@ class product_attribute_line(osv.osv):
         'value_ids': fields.many2many('product.attribute.value', id1='line_id', id2='val_id', string='Product Attribute Value'),
     }
 
+    def _check_valid_attribute(self, cr, uid, ids, context=None):
+        obj_pal = self.browse(cr, uid, ids[0], context=context)
+        if any(v in obj_pal.attribute_id.value_ids.ids for v in obj_pal.value_ids.ids):
```

Comment:
I wasn't referring to a risk of false positive but rather to a false negative.
Indeed I'm reading his code like this: "It's ok if at least one value is valid".
If I'm not mistaken, _all_ line values should be correct?